tools/man/ck2cti.1
.TH "ck2cti" 1 "16 Nov 2003" "ck2cti" \" -*- nroff -*-
.ad l
.nh
.SH NAME
ck2cti \- convert a Chemkin-II-format reaction mechanism file to Cantera input format.  

.SH SYNOPSIS
.br
ck2cti \fB-i \fIinput \fR[\fB-t \fIthermo\fR] \fR[\fB-tr\fI transport\fR] 
\fR[\fB-id\fI idtag\fR]

.SH DESCRIPTION

.I ck2cti
converts a Chemkin-II-format reaction mechanism file to Cantera input format. 
The resulting Cantera input file contains a definition of one 
.B ideal_gas
entry that represents an ideal gas mixture corresponding to the Chemkin-II reaction mechanism. The file also contains Cantera-format definitions for each species and each reaction in the input reaction mechanism file.
The Cantera-format text is written to the standard output.

The parameters are as follows:
.TP
.I input
Chemkin-II reaction mechanism file to be converted. Required.
.TP
.I thermo
Thermodynamic property database. If the THERMO section of the input file is missing or does not have entries for one or more species, this file will be searched for the required thermo data. This file may be another reaction mechanism file containing a THERMO section, or a Chemkin-II-compatible thermodynamic database file.
.TP
.I transport
Transport property database. If this file name is supplied, transport property parameters will be taken from this file and included in the output Cantera-format file. If this parameter is omitted, no transport property parameters will be included in the output.
.TP
.I id
Identification string. The \fBideal_gas\fR entry in the Cantera-format output has name  \fIid\fR. If this parameter is omitted, it will be set to the input file name without the extension. Since only one phase definition is present in the \fIck2cti\fR output, this parameter is not required.

.SH EXAMPLES

Create a Cantera input file from a self-contained Chemkin-format file
without transport parameters:
.TP
ck2cti -i mech.inp > mech.cti
.TP
Create a Cantera input file from a Chemkin-format file that requires a
separate thermodynamic database
without transport parameters:
.TP
ck2cti -i mech.inp -t therm.dat > mech.cti
.TP
Create a Cantera input file from a Chemkin-format file that requires a
separate thermodynamic database and include transport parameters:
.TP
ck2cti -i mech.inp -t therm.dat -tr tran.dat > mech.cti
.TP



ChangeLog
2006-11-14 00:40  dggoodwin

	* configure, configure.in, Cantera/src/MultiPhaseEquil.cpp: minor
	  cleanup

2006-11-13 11:55  dggoodwin

	* Cantera/src/CVodesIntegrator.h, ext/f2c_libs/arith.h,
	  win32/vc7/Sundials/CVODES/CVODES.vcproj,
	  win32/vc7/Sundials/NVEC_SER/NVEC_SER.vcproj,
	  win32/vc7/Sundials/SUNDIALS_SHARED/SUNDIALS_SHARED.vcproj,
	  win32/vc7/cantera/cantera.vcproj: minor cleanup

2006-11-13 11:46  dggoodwin

	* win32/vc7/Sundials/sundials/sundials_config.h: added
	  sundials_config.h so that users can build sundials without having
	  to run configure

2006-11-13 08:39  dggoodwin

	* ChangeLog, preconfig, ext/f2c_libs/arith.h,
	  test_problems/python/runtest: minor changes

2006-11-12 20:23  dggoodwin

	* License.rtf, preconfig, winconfig.h, ext/f2c_libs/arith.h: minor
	  cleanup

2006-11-12 20:22  dggoodwin

	* win32/vc7/demos/: NASA_coeffs.vcproj, combustor.vcproj,
	  flamespeed.vcproj, kinetics1.vcproj: added example project files

2006-11-12 20:21  dggoodwin

	* win32/vc7/: SetupCantera/SetupCantera.vdproj,
	  SetupCantera/SetupCanteraLite.vdproj, cantera.sln,
	  Sundials/CVODES/CVODES.vcproj,
	  Sundials/CVODES_DENSE/CVODES_DENSE.vcproj,
	  Sundials/IDA/IDA.vcproj, Sundials/KINSOL/KINSOL.vcproj,
	  Sundials/NVEC_SER/NVEC_SER.vcproj,
	  Sundials/SUNDIALS_SHARED/SUNDIALS_SHARED.vcproj,
	  cantera/cantera.vcproj, demos/CanteraDemos.sln,
	  demos/Rankine.vcproj, demos/demo.vcproj, zeroD/zeroD.vcproj:
	  update of project files

2006-11-11 06:12  dggoodwin

	* win32/vc7/: cantera.sln, cantera_no_sundials.sln,
	  SetupCantera/SetupCantera.vdproj, cantera/cantera.vcproj: added
	  no_sundials version

2006-11-11 06:10  dggoodwin

	* win32/vc7/Sundials/: Sundials.sln,
	  CVODES_DENSE/CVODES_DENSE.vcproj: added cvodes_dense

2006-11-11 05:23  dggoodwin

	* preconfig, Cantera/src/Makefile.in,
	  Cantera/src/zeroD/Makefile.in: minor changes to how dependencies
	  are handled in Makefiles

2006-11-10 17:38  dggoodwin

	* configure, configure.in: fixed null username

2006-11-10 17:37  dggoodwin

	* tools/src/finish_install.py.in: added writing PYTHONPATH to
	  setup_Cantera for min_Python

2006-11-10 05:48  dggoodwin

	* Cantera/src/CVodesIntegrator.h: minor cleanup

2006-11-09 05:04  dggoodwin

	* Cantera/python/examples/fuel_cells/: sofc.cti, sofc.py: added
	  fuel cell demo

2006-11-08 16:19  dggoodwin

	* Cantera/src/ThermoPhase.h: minor cleanup

2006-11-08 15:58  dggoodwin

	* Makefile.in, preconfig, Cantera/python/Cantera/OneD/onedim.py,
	  ext/f2c_libs/arith.h: install Makefiles for python demos

2006-11-08 15:14  dggoodwin

	* Cantera/python/examples/: Makefile, equilibrium/Makefile,
	  equilibrium/stoich.py, flames/Makefile, gasdynamics/Makefile,
	  kinetics/Makefile, liquid_vapor/Makefile, misc/Makefile,
	  reactors/Makefile, surface_chemistry/Makefile,
	  transport/Makefile: added Makefiles to run all demos

2006-11-07 13:03  dggoodwin

	* Cantera/src/zeroD/ReactorNet.h: added verbose, setVerbose

2006-11-07 13:02  dggoodwin

	* Cantera/src/zeroD/ReactorBase.h: added setName

2006-11-07 05:47  dggoodwin

	* Cantera/src/: State.h, State.cpp: added method concentration(int
	  k)

2006-11-06 07:27  dggoodwin

	* Cantera/cxx/demos/NASA_coeffs.cpp: added NASA polynomial example

2006-11-06 07:20  dggoodwin

	* Cantera/src/: Constituents.h, NasaPoly1.h, NasaThermo.h,
	  SpeciesThermo.h, SpeciesThermoFactory.cpp,
	  SpeciesThermoInterpType.h: added support for modifying NASA
	  polynomial coefficients

2006-11-05 07:54  dggoodwin

	* data/inputs/methane_pox_on_pt.cti: added mechanism for partial
	  oxidation of methane on platinum from the Deutschmann group

2006-11-03 12:45  dggoodwin

	* winconfig.h: updated

2006-11-03 12:44  dggoodwin

	* win32/vc7/Sundials/: CVODES/CVODES.vcproj, IDA/IDA.vcproj,
	  KINSOL/KINSOL.vcproj, NVEC_SER/NVEC_SER.vcproj,
	  SUNDIALS_SHARED/SUNDIALS_SHARED.vcproj: initial import

2006-11-02 17:32  dggoodwin

	* win32/vc7/SetupCantera/SetupCantera.vdproj,
	  win32/vc7/SetupCantera/SetupCanteraLite.vdproj,
	  win32/vc7/ctmatlab/runmlab.cmd, Cantera/python/winsetup.py,
	  License.txt: updated for Cantera 1.7

2006-10-31 11:09  dggoodwin

	* Cantera/src/ReactionStoichMgr.cpp: fixed bug that resulted in
	  incorrect computation of product concentration products when rxn
	  has 4 or more product molecules and one or more has a stoich
	  coeff > 1

2006-10-30 11:30  dggoodwin

	* configure, configure.in, preconfig, ext/f2c_libs/arith.h,
	  tools/src/finish_install.py.in: modified for cygwin

2006-10-27 13:34  dggoodwin

	* configure, configure.in, tools/src/finish_install.py.in: mods for
	  cygwin

2006-10-22 18:13  dggoodwin

	* Cantera/src/CVodesIntegrator.cpp: sundials version detection

2006-10-22 17:58  dggoodwin

	* preconfig: debug off is noe the default

2006-10-22 17:56  dggoodwin

	* Cantera/src/: ChemEquil.cpp, MultiPhaseEquil.cpp:  moved
	  declaration of nCutoff to avoid jumping over it, causing a
	  compile failure with gcc 4.0.1 on linuxChemEquil.cpp

2006-10-22 17:51  dggoodwin

	* tools/templates/cxx/demo.mak.in: removed FLIBS

2006-10-20 14:56  hkmoffa

	* test_problems/min_python/Makefile.in: Added -s to make tests.

2006-10-20 14:55  hkmoffa

	* test_problems/cathermo/HMW_graph_GvI/.cvsignore: Added a file.

2006-10-20 14:54  hkmoffa

	* test_problems/fracCoeff/runtest: Added an exit variable.

2006-10-20 14:47  hkmoffa

	* test_problems/ChemEquil_gri_matrix/output_blessed.txt: Slight
	  change to the blessed solution.

2006-10-20 14:43  hkmoffa

	* configure, configure.in, test_problems/Makefile.in: Added new
	  test directories.

2006-10-20 14:42  hkmoffa

	* Cantera/src/sort.h: Added an Id stamp.

2006-10-20 14:41  hkmoffa

	* Cantera/src/misc.cpp: Fixed an error in the LogGroup system.

2006-10-20 14:40  hkmoffa


tools/src/ck2cti.cpp
/**
 * @file ck2cti.cpp
 * $Id: ck2cti.cpp,v 1.6 2005/06/18 17:01:38 dggoodwin Exp $
 *
 * Program to convert Chemkin-II-format reaction mechanism files to
 * Cantera input format. The resulting Cantera input file contains a
 * definition of one ideal_gas entry that represents an ideal gas
 * mixture corresponding to the Chemkin-II reaction mechanism.  The
 * file also contains Cantera-format definitions for each species and
 * each reaction in the input reaction mechanism file.
 *
 * Usage: ck2cti -i input -t thermo -tr transport -id idtag
 *
 * The Cantera-format text is written to the standard output.
 *
 * @param input Chemkin-II reaction mechanism file to be converted. Required.
 *
 * @param thermo Thermodynamic property database. If the THERMO section of the 
 * input file is missing or does not have entries for one or more species,
 * this file will be searched for the required thermo data. This file may 
 * be another reaction mechanism file containing a THERMO section, or
 * a Chemkin-II-compatible thermodynamic database file.
 *
 * @param transport Transport property database. If this file name is supplied,
 * transport property parameters will be taken from this file and 
 * included in the output Cantera-format file. If this parameter is omitted, 
 * no transport property parameters will be included in the output.
 *
 * @param id idtag. The ideal_gas entry in the Cantera-format output 
 * has name \i idtag. If this parameter is omitted, it will be set to the
 * input file name without the extension. Since only one phase definition
 * is present in the ck2cti output, this parameter is not required.  
 */
#ifdef WIN32
#pragma warning(disable:4786)
#pragma warning(disable:4503)
#endif

#include <iostream>
#include <string>
using namespace std;

#include "ct_defs.h"
#include "global.h"
#include "converters/ck2ct.h"

using namespace Cantera;

int showHelp() {
    cout << "\nck2cti: convert a CK-format reaction mechanism file to Cantera input format.\n"
         << "\n   D. G. Goodwin, Caltech \n"
         << "   Version 1.0, August 2003.\n\n"
         << endl;
    cout << "options:" << endl;
    cout << "    -i  <input file> \n"
         << "    -t  <thermo database> \n"
         << "    -tr <transport database> \n"
         << "    -id  <identifier> \n"
         << "    -d  print debugging output \n\n"
         << "    -v  validate the input file \n\n"
         << "The results are written to the standard output.\n";
    return 0;
}

string getp(int& i, int argc, char** args) {
    string a="--";
    if (i < argc-1) {
        a = string(args[i+1]);
    }
    if (a[0] == '-') {
        a = "<missing>";
    }
    else {
        i += 1;
    }
    return a;
}


int main(int argc, char** argv) {
    string infile="chem.inp", dbfile="", trfile="", logfile;
    string idtag = "gas";
    bool debug = false;
    bool validate = false;
    int i=1;
    if (argc == 1) return showHelp();
 
    while (i < argc) {
        string arg = string(argv[i]);
        if (arg == "-i") {
            infile = getp(i,argc,argv);
        }
        else if (arg == "-t") {
            dbfile = getp(i,argc,argv);;
        }
        else if (arg == "-tr") {
            trfile = getp(i,argc,argv);
        }
        else if (arg == "-id") {
            idtag = getp(i,argc,argv);
        }
        else if (arg == "-d") {
            debug = true;
            cout << "### DEBUG MODE ###" << endl;
        }
        else if (arg == "-v") {
            validate = true;
            cout << "### VALIDATION ENABLED ###" << endl;
        }
        else if (arg == "-h" || argc < 3) {
            return showHelp(); 
        }
        else {
            cout << "unknown option:" << arg << endl;
            exit(-1);
        }
        ++i;
    }

    int ierr = pip::convert_ck(infile.c_str(), dbfile.c_str(), trfile.c_str(), 
        idtag.c_str(), debug, validate);

    if (ierr < 0) {
        showErrors(cerr);
    }
    return ierr;
}
